packages/components/switch/src/switch.ts

```diff
@@ -0,0 +1,164 @@
+import type { IElementInternals } from 'element-internals-polyfill';
+import type { CSSResultGroup, PropertyValues, TemplateResult } from 'lit';
+import type { EventEmitter } from '@sl-design-system/shared';
+import {
+  EventsController,
+  FormControlMixin,
+  HintMixin,
+  ValidationController,
+  event,
+  requiredValidator,
+  validationStyles
+} from '@sl-design-system/shared';
+import type { IconSize } from '@sl-design-system/icon';
+import { LitElement, html } from 'lit';
+import { property, queryAssignedElements } from 'lit/decorators.js';
+import styles from './switch.scss.js';
+
+export type SwitchSize = 'sm' | 'md' | 'lg';
+export type SwitchOrientation = 'horizontal' | 'vertical';
+
+export class Switch extends FormControlMixin(HintMixin(LitElement)) {
+  /** @private */
+  static formAssociated = true;
+
+  /** @private */
+  static override styles: CSSResultGroup = [validationStyles, styles];
```

Comment:
Maybe we should add `hintStyles` as well and use 'hintsize' like in other forms controls like textarea etc. to have it consistent? And we will not need to use '--_hint-font' in switch.scss file?

Reply:
done. needed to overwrite the padding-top though, because in the switch it doesn't have any padding.